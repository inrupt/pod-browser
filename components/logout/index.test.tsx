import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Router from "next/router";
import { logout } from "@inrupt/solid-auth-fetcher";

import LogOutButton from "./index";

jest.mock("next/router");
jest.mock("@inrupt/solid-auth-fetcher");

describe("Logout button", () => {
  test("Renders a logout button", () => {
    const tree = shallow(<LogOutButton />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Calls logout and redirects on click", () => {
    const tree = shallow(<LogOutButton />);
    tree.simulate("click", { preventDefault: () => {} });

    expect(Router.push).toHaveBeenCalledWith("/login");
    expect(logout).toHaveBeenCalled();
  });
});
