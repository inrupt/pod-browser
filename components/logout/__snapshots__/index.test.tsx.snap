// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout button Renders a logout button 1`] = `
<WithStyles(ForwardRef(Button))
  onClick={[Function]}
>
  Log out
</WithStyles(ForwardRef(Button))>
`;
